### ISSUE TITLE
train 2 unet error 

### ISSUE THREAD

**Issue by FTKyaoyuan** [lucidrains/imagen-pytorch#241]
hello 
I want to try to train two unet 
this is my  train code 
```
def train(img):
    global LOSS
    Loss=[0,0]
    for i in range(2):
        name = "image_checkpoint.pth"
        trainer.load(name)
        loss = trainer(img,unet_number =i+1)
        Loss[i] = Loss[i]+loss
        trainer.update(unet_number=i+1)
        trainer.save(name)
    wandb.log({"u1_loss":Loss[0],"u2_loss":Loss[1]},step=step)
    LOSS[0]=LOSS[0]+Loss[0]
    LOSS[1]=LOSS[1]+Loss[1]
```
but an error occurs

`you cannot only train on one unet at a time. you will need to save the trainer into a checkpoint, and resume training on a new unet`

Therefore I changed my code

```
def train(img):
    global LOSS
    Loss=[0,0]
    for i in range(2):
        name = "image_checkpoint"+str(i+1)+".pth"
        trainer.load(name)
        loss = trainer(img,unet_number =i+1)
        Loss[i] = Loss[i]+loss
        trainer.update(unet_number=i+1)
        trainer.save(name)
    wandb.log({"u1_loss":Loss[0],"u2_loss":Loss[1]},step=step)
    LOSS[0]=LOSS[0]+Loss[0]
    LOSS[1]=LOSS[1]+Loss[1]
```
But he still got the above error

How to train two unet ？
thank

**Comment by turboLJY** [comment]
Hi, I got the same error. Did you solve this problem? thank

**Comment by jacobwjs** [comment]
Some suggestions.

Loading and saving each unet for every image seems like a good way to slow down training.

Try training a single unet for all your images and for many epochs. Then repeat for the next upscaled version.

**Comment by turboLJY** [comment]
Oh I see, thank~

**Comment by dinhdungz** [comment]
@jacobwjs i had training with one unet for many epochs. Then save and training with 2 unet but i have same error
for i in (1, 2):
  l_arr = []
  for eps in range(1, epochs+1):
      with tqdm(total=len(batches)) as pbar:
          for step, e in enumerate(batches[:200]):
              
              batch = batches[step]

              text_embeds, text_masks = get_emb_tensor(texts, batch_ss, step)
              images = get_images_tensor(batch)

              if meth == 'imagen':
                loss = imagen(images, text_embeds = text_embeds, text_masks = text_masks, unet_number = i)
                loss.backward()
              else:
                loss = trainer(images, text_embeds = text_embeds, text_masks = text_masks, unet_number = i, max_batch_size = batch_s)
                trainer.update(unet_number = i)
              l_arr.append(loss)
              if step % checkpoint_rate == 0 and step !=0 and not math.isnan(loss):
                  trainer.save(f'./checkpoint.pt')
              pbar.set_description(f'Unet Num: {i}  Epoch: {eps}  Loss: {loss} Avg Loss: {round((sum(l_arr)/len(l_arr)), 3)}')
              pbar.update()
      if not math.isnan(loss): trainer.save(f'./checkpoint.pt')

**Comment by TheFusion21** [comment]
afaik you have to reinstantiate the trainer and/or imagen and then load the cp

**Comment by dinhdungz** [comment]
@TheFusion21 Thanks you, that error is gone
